src/algebra/order/floor.lean

```diff
@@ -214,6 +214,68 @@ ext $ λ x, ceil_eq_iff hn
 
 /-! #### Intervals -/
 
+lemma floor_mem_Icc_of_Icc {a b : ℕ} {x : α} (hx : x ∈ set.Icc (a : α) ↑b) :
+  ↑⌊x⌋₊ ∈ set.Icc (a : α) ↑b :=
+begin
+  rw set.mem_Icc,
+  norm_cast,
+  refine ⟨le_floor hx.left, _⟩,
+  have := floor_mono hx.right,
+  rwa floor_coe at this,
+end
+
+lemma floor_mem_Icc_of_Ico {a b : ℕ} {x : α} (hx : x ∈ set.Ico (a : α) ↑b) :
```

Comment:
the conclusion should be that it belongs to `Ico`, which is stronger. Same thing in some other lemmas, can you try to optimize the conclusion?